public/components/trace_analytics/components/traces/span_detail_table.tsx

```diff
@@ -261,37 +252,248 @@ export function SpanDetailTable(props: SpanDetailTableProps) {
+    <EuiButtonEmpty
+      size="xs"
+      onClick={fullScreenMode ? closeFullScreenModal : openFullScreenModal}
+      key="fullScreen"
+      color="text"
+      iconType={fullScreenMode ? 'cross' : 'fullScreen'}
+      data-test-subj="fullScreenButton"
+    >
+      {fullScreenMode ? 'Exit full screen' : 'Full screen'}
+    </EuiButtonEmpty>,
+  ];
+
   return (
     <>
-      <EuiDataGrid
-        aria-labelledby="span-detail-data-grid"
-        columns={columns}
-        columnVisibility={{ visibleColumns, setVisibleColumns }}
-        rowCount={total}
-        renderCellValue={renderCellValue}
-        sorting={mode === 'jaeger' ? undefined : { columns: tableParams.sortingColumns, onSort }}
-        toolbarVisibility={mode === 'jaeger' ? false : true}
-        pagination={{
-          pageIndex: tableParams.page,
-          pageSize: tableParams.size,
-          pageSizeOptions: [10, 50, 100],
-          onChangeItemsPerPage,
-          onChangePage,
-        }}
-      />
+      <FullScreenWrapper isFullScreen={fullScreenMode} onClose={closeFullScreenModal}>
+        <EuiDataGrid
+          aria-labelledby="span-detail-data-grid"
+          columns={columns}
+          columnVisibility={{ visibleColumns, setVisibleColumns }}
+          rowCount={total}
+          renderCellValue={renderCellValue}
+          sorting={mode === 'jaeger' ? undefined : { columns: tableParams.sortingColumns, onSort }}
+          toolbarVisibility={{
+            showColumnSelector: true,
+            showSortSelector: true,
+            showFullScreenSelector: false,
+            additionalControls: toolbarButtons,
+          }}
+          pagination={{
+            pageIndex: tableParams.page,
+            pageSize: tableParams.size,
+            pageSizeOptions: [10, 50, 100],
+            onChangeItemsPerPage,
+            onChangePage,
+          }}
+          style={{
+            height: fullScreenMode ? '100%' : 'auto',
+          }}
+        />
+      </FullScreenWrapper>
       {total === 0 && <NoMatchMessage size="xl" />}
     </>
   );
 }
+
+export function SpanDetailTableHierarchy(props: SpanDetailTableProps) {
+  const { mode } = props;
+  const [items, setItems] = useState<any>([]);
+  const [total, setTotal] = useState(0);
+  const [expandedRows, setExpandedRows] = useState(new Set<string>());
+  const [visibleColumns, setVisibleColumns] = useState<string[]>([]);
+
+  useEffect(() => {
+    const spanSearchParams = {
+      from: 0,
+      size: 10000,
+      sortingColumns: [],
+    };
+    handleSpansRequest(
+      props.http,
+      (data) => {
+        const hierarchy = buildHierarchy(data);
+        setItems(hierarchy);
+      },
+      setTotal,
+      spanSearchParams,
+      props.DSL,
+      mode,
+      props.dataSourceMDSId
+    );
+  }, [props.DSL]);
+
+  interface Span {
+    spanId: string;
+    parentSpanId?: string;
+    children: Span[];
+    [key: string]: any;
+  }
+
+  type SpanMap = Record<string, Span>;
```

Comment:
nit. Not blocker if these are re-usable contants, we can move these to https://github.com/opensearch-project/dashboards-observability/blob/main/public/components/common/types.ts